components/bootloader_support/src/esp_image_format.c

```diff
@@ -398,13 +425,28 @@ static esp_err_t process_segment(int index, uint32_t flash_addr, esp_image_segme
 
 static esp_err_t process_segment_data(intptr_t load_addr, uint32_t data_addr, uint32_t data_len, bool do_load, bootloader_sha256_handle_t sha_handle, uint32_t *checksum)
 {
+    // If we are not loading, and the checksum is empty, skip processing this
+    // segment for data
+    if(!do_load && NULL == checksum)
```

Comment:
Same suggestion applies here. This function already has a lot of arguments but testing `if(!do_load && !do_verify)` is be more readable. Same applies in other places.

Reply:
I'm just realizing this shortcut to skipping an unneeded mmap() invocation probably should also check that `sha_handle != NULL` as well.  (That way it's possible to use the function for a SHA check without checksum.)  The only point this check is currently hit is with the 

Yeah... there are a lot of arguments to this function, and to process_segment(), too.  Do_load/do_verify could be made into single parameter via an enum/bitmask along the lines of:

typedef enum {
    ESP_SEGMENT_DO_NOTHING = 0,
    ESP_SEGMENT_DO_LOAD         = 1,
    ESP_SEGMENT_DO_VERIFY      = 2,
    ESP_SEGMENT_DO_LOAD_DO_VERIFY = 3,
} esp_process_segment_opt_t;

Then the check could be `if (ESP_SEGMENT_DO_NOTHING == operation)`.  Although, having a pointer that is NULL also (at least to me) says that it's not intended to be used.  Granted, it's a little more obvious what it is doing with an intentional desired operation parameter passed along.

-- Tim